File: python/leetcode/1004-max-consecutive-ones-III.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:

        longest = 0
        num_zeroes = 0
        left = 0

        for right in range(len(nums)):
            if nums[right] != 1:
                num_zeroes += 1

            while num_zeroes > k:
                if nums[left] != 1:
                    num_zeroes -= 1
                left += 1

            longest = max(longest, right-left+1)

        return longest

File: c#/leetcode/1004-max-consecutive-ones-iii.cs
public class Solution
{
  public int LongestOnes(int[] nums, int k)
  {
    int left = 0;
    int longest = 0;
    int zeroes = 0;

    for (int right = 0; right < nums.Length; right++)
    {
      if (nums[right] == 0)
      {
        zeroes++;
      }

      while (zeroes > k)
      {
        if (nums[left] == 0)
        {
          zeroes--;
        }
        left++;
      }

      longest = Math.Max(longest, right - left + 1);
    }
    return longest;
  }
}